geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -1885,10 +1886,8 @@ public boolean executeWith(Object entList, int b) {
           if (lostMemberVersionID == null) {
             lostMemberVersionID = lostMemberID;
           }
-          // check to see if the region in this cache needs to synchronize with others
-          // it is possible that the cache is recover/restart of a member and not
-          // scheduled to synchronize with others
-          synchronizeIfNotScheduled(targetRegion, lostMemberID, lostMemberVersionID);
+          waitInAnotherThreadToCheckIfSynchronizationScheduled(targetRegion, lostMemberID,
```

Comment:
Could check the isRegionSynchronizeScheduledOrDone condition here. If so we do not need to wait.
if (!targetRegion.isRegionSynchronizeScheduledOrDone) {
  waitInAnotherThreadToCheckIfSynchronizationScheduled(targetRegion, lostMemberID, lostMemberVersionID);
}

Reply:
I would rather let your original code to handle that to reduce code changes.